### PR TITLE
Feat/add matomo language button tracking

### PR DESCRIPTION
**Pull request by aeschi**
we also need to change NEXT_PUBLIC_MATOMO_URL in .env

https://app.asana.com/0/1204152181784036/1206612990757854

### CODE REVIEW

**Comment by aeschi** [comment]
I changed .env for staging accordingly

File: src/components/Switch/matomo.ts
Comment (raphael-arce) [inline src/components/Switch/matomo.ts:1]: let's move this file in a more global folder. maybe in the `utils` folder 👍

**Review by raphael-arce — APPROVED** [review]
one suggestion otherwise lgtm 👍  thx

**Comment by raphael-arce** [comment]
@aeschi @ff6347 @Jaszkowic should we test this in some manner?

File: src/components/Switch/matomo.ts
Comment (aeschi) [reply src/components/Switch/matomo.ts:1]: done!

**Comment by ff6347** [comment]
But how? Intercept the request with MSW could be one solution.

**Review by ff6347 — CHANGES_REQUESTED** [review]
Great you made this work with Matomo. Good job.

My 50cent on the structure:

- Should a type declaration on the global window object be buried in a utils file? Shouldn't we introduce something like `src/common/globals.d.ts` or `src/common/window.d.ts` for things like this?
- Also file in `/util` holds a function that is solely used in one other location. The `Switch/index.tsx` Shouldn't it be located there instead of introducing a new utils file with one single not reused function?
- The type of `any[]` is only there to shut Typescript up. If you want TS to shut up a `//@ts-ignore` is enough. If you want type safety us `unknown` and do some defensive programming please.

**Comment by raphael-arce** [comment]
@ff6347 I‘m joining the discussion as I pair-programmed with @aeschi to implement this. 

1. yes, that‘s a valid point.
2. while this function is only used once atm, we expect it to be used more often in the future to track specific events (e.g. tracking on salespage outlink), that‘s why I suggested to move it in a more global folder. Would you still prefer to have moved while there is only a single usage?
3. valid point as well!

regarding the testing, i agree, we should use something like msw. However, I researched a bit and found this feature from playwright: https://playwright.dev/docs/mock 

especially the HAR files feature seem interesting to mock the analytics script from matomo. What do you think?

**Comment by ff6347** [comment]
> @ff6347 I‘m joining the discussion as I pair-programmed with @aeschi to implement this.
 
> 2. while this function is only used once atm, we expect it to be used more often in the future to track specific events (e.g. tracking on salespage outlink), that‘s why I suggested to move it in a more global folder. Would you still prefer to have moved while there is only a single usage?

I guess _when_ we start using this in other locations we still can refactor the code base to move it into another shared file. Having 3 lines of code in a file just in case somewhere in the future we might reuse it, is (IMHO), over engineering.


> regarding the testing, i agree, we should use something like msw. However, I researched a bit and found this feature from playwright: https://playwright.dev/docs/mock
> 
> especially the HAR files feature seems interesting to mock the analytics script from matomo. What do you think?

If we plan to use playwright for more e2e tests I would say let's go for playwright. We already have msw in the code base I would not introduce another dep just for testing this one feature. So currently I would see if we can do this with msw first.

**Comment by raphael-arce** [comment]
@ff6347 I tried to render the LandingPage via vitest and it looks like it is going to be hard to test the matomo tracking with the means we currently have:
- React Testing Library ignores script tags: 
Bildschirmfoto 2024-02-19 um 20 16 11

- And apparently, you also cannot render the Head component from nextjs: https://github.com/testing-library/react-testing-library/issues/402#issuecomment-1013392720  (links to this discussion https://github.com/vercel/next.js/discussions/11060)
- Vitest has an experimental browser mode https://vitest.dev/guide/browser.html, I tried to use it but got an error message that vitest 1.3.0 is necessary, right now we have 1.2.1.

Maybe with more research there is a way to make it work. But probably it is going to be easier to use playwright straight away, once we decide to use it. Let's keep it in mind for the UI rework 👍

**Review by ff6347 — APPROVED** [review]

**Pull request merged**